task/control-plane-agent/src/main.rs

```diff
@@ -315,6 +312,89 @@ impl idl::InOrderControlPlaneAgentImpl for ServerImpl {
 
         Ok(id)
     }
+
+    #[cfg(feature = "gimlet")]
+    fn get_uart_client(
```

Comment:
Optional: I usually put groups of operations like this into a separate `impl` block, to avoid copy-pasting the `#[cfg(feature...)]` many times.

Reply:
Hmm, I don't think I can do this here, because these methods are required by `idl::InOrderControlPlaneAgentImpl`. Some options:

1. Leave it as-is
2. Split out separate `ServerImpl::get_uart_client_impl()` etc methods (that can be defined in separate impl blocks), then call them from here
3. Add methods to `mgs_sidecar`+`mgs_psc` so we can always call `self.mgs_handler.uart_client()` etc from here (i.e., no features required), and have those return errors instead

I think my preference of those is 1,3,2, but it's not particularly strong. What do you think?

Reply:
Huh, I thought that you could split the trait impl into multiple blocks (some gated with features), but it looks like that's not possible:
```
error[E0119]: conflicting implementations of trait `idl::InOrderControlPlaneAgentImpl` for type `ServerImpl`
   --> task/control-plane-agent/src/main.rs:318:1
    |
258 | impl idl::InOrderControlPlaneAgentImpl for ServerImpl {
    | ----------------------------------------------------- first implementation here
...
318 | impl idl::InOrderControlPlaneAgentImpl for ServerImpl {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `ServerImpl`

For more information about this error, try `rustc --explain E0119`.
error: could not compile `task-control-plane-agent` due to previous error
Error: failed to build control_plane_agent

Caused by:
    command failed, see output for details
```

In that case, I'm fine leaving this as-is.